castervoice/apps/dragon.py

```diff
@@ -71,44 +72,56 @@ class DragonRule(MergeRule):
         "right point":
             R(Playback([(["MouseGrid"], 0.1), (["six", "six"], 0.1), (["click"], 0.0)])),
         "center point":
-            R(Playback([(["MouseGrid"], 0.1), (["click"], 0.0)])),
-        "show windows":
-            R(Mimic("list", "all", "windows")),
-        "cory <text>":
-            R(Mimic("correct", extra="text") + WaitWindow(title="spelling window") +
-                Mimic("choose", "one")),
-        "cory that":
-            R(Mimic("correct", "that") + WaitWindow(title="spelling window") +
-                Mimic("choose", "one")),
-        "make that <text>":
-            R(Mimic("scratch", "that") + Mimic(extra="text")),
-        "scratch [<n10>]":
-            R(Playback([(["scratch", "that"], 0.03)])),
-        "train word":
-            R(Mimic("train", "that") + Key("a-r/200, s")),
-        "word train":
-            R(Key("c-c/20") + Mimic("edit", "vocabulary") + Pause("100") +
-                Key("c-v/5, tab, down, up, a-t/50, enter/50, a-r/250, s/50, escape")),
-        "(add train | train from add word)":
-            R(Key("a-a/2, enter/300, a-s")),
-        # Users may want to adjust the way time on the next four commands
-        "(train from vocab | cab train)":
-            R(Key("a-t/50, enter/50, a-r/250, s")),
-        "(train from vocab | cab train)":
-            R(Key("a-t/50, enter/50, a-r/250, s")),
+            R(Playback([(["MouseGrid"], 0.1), (["click"], 0.0)]),
+              rdescript="Mouse: Center Point"),
+        
+        
+        "show windows": R(Mimic("list", "all", "windows"),  
+            rdescript="Dragon: emulate Dragon command for listing windows"),
+        "cory <text>": 
+            R(Mimic("correct", extra="text") + WaitWindow(title="spelling window") + Mimic("choose", "one"),
+                rdescript="Dragon: brings up the correction menu for the phrase spoken in the command and chooses the 1st choice"),
+        "cory that": 
+            R(Mimic("correct", "that") + WaitWindow(title="spelling window") + Mimic("choose", "one"), 
+                rdescript="Dragon: brings up the correction menu for the previously spoken phrase and chooses the first choice"),
+
+        "make that <text>": R(Mimic("scratch", "that") + Mimic(extra="text"), 
+             rdescript="Dragon: deletes the dictation generated by the previous utterance and replaces it with what you say next"),
+        "scratch [<n10>]": R(Playback([(["scratch", "that"], 0.03)]), 
+            rdescript="Dragon: delete dictation from previous n utterances") * Repeat(extra="n10"),
+
+            # Users may want to adjust the wait time on the next few commands     
+        "train word": R(Mimic("train", "that") + Pause("75") + Key("a-r/250, s"),
```

Comment:
I don't think "train that" is a valid DNS command as it prints "train that".

Reply:
You have to be in a full text control application for the command "train that" to work